src/main.cc

```diff
@@ -1,131 +1,139 @@
+    Napi::TypeError::New(env, "Parameter 'service' must be a string").
+      ThrowAsJavaScriptException();
+    return env.Null();
   }
 
-  Nan::Utf8String serviceNan(info[0]);
-  std::string service(*serviceNan, serviceNan.length());
+  std::string service = info[0].As<Napi::String>();
 
-  if (!info[1]->IsString()) {
-    Nan::ThrowTypeError("Parameter 'username' must be a string");
-    return;
+  if (!info[1].IsString()) {
+    Napi::TypeError::New(env, "Parameter 'username' must be a string").
+      ThrowAsJavaScriptException();
+    return env.Null();
   }
 
-  Nan::Utf8String usernameNan(info[1]);
-  std::string username(*usernameNan, usernameNan.length());
+  std::string username = info[1].As<Napi::String>();
 
-  if (!info[2]->IsString()) {
-    Nan::ThrowTypeError("Parameter 'password' must be a string");
-    return;
+  if (!info[2].IsString()) {
+    Napi::TypeError::New(env, "Parameter 'password' must be a string").
+      ThrowAsJavaScriptException();
+    return env.Null();
   }
 
-  Nan::Utf8String passwordNan(info[2]);
-  std::string password(*passwordNan, passwordNan.length());
+  std::string password = info[2].As<Napi::String>();
 
   SetPasswordWorker* worker = new SetPasswordWorker(
     service,
     username,
     password,
-    new Nan::Callback(info[3].As<v8::Function>()));
-  Nan::AsyncQueueWorker(worker);
+    env);
+  worker->Queue();
+  return worker->Promise();
 }
 
-NAN_METHOD(GetPassword) {
-  if (!info[0]->IsString()) {
-    Nan::ThrowTypeError("Parameter 'service' must be a string");
-    return;
+Napi::Value GetPassword(const Napi::CallbackInfo& info) {
+  Napi::Env env = info.Env();
+  if (!info[0].IsString()) {
+    Napi::TypeError::New(env, "Parameter 'service' must be a string").
+      ThrowAsJavaScriptException();
+    return env.Null();
   }
 
-  Nan::Utf8String serviceNan(info[0]);
-  std::string service(*serviceNan, serviceNan.length());
+  std::string service = info[0].As<Napi::String>();
 
-  if (!info[1]->IsString()) {
-    Nan::ThrowTypeError("Parameter 'username' must be a string");
-    return;
+  if (!info[1].IsString()) {
+    Napi::TypeError::New(env, "Parameter 'username' must be a string").
+      ThrowAsJavaScriptException();
+    return env.Null();
   }
 
-  Nan::Utf8String usernameNan(info[1]);
-  std::string username(*usernameNan, usernameNan.length());
+  std::string username = info[1].As<Napi::String>();
 
   GetPasswordWorker* worker = new GetPasswordWorker(
     service,
     username,
-    new Nan::Callback(info[2].As<v8::Function>()));
-  Nan::AsyncQueueWorker(worker);
+    env);
+  worker->Queue();
+  return worker->Promise();
 }
 
-NAN_METHOD(DeletePassword) {
-  if (!info[0]->IsString()) {
-    Nan::ThrowTypeError("Parameter 'service' must be a string");
-    return;
+Napi::Value DeletePassword(const Napi::CallbackInfo& info) {
+  Napi::Env env = info.Env();
+  if (!info[0].IsString()) {
+    Napi::TypeError::New(env, "Parameter 'service' must be a string");
```

Comment:
Is `ThrowAsJavaScriptException` missing here?

Reply:
I think it makes sense to extract a method like that:

```
void ThrowTypeError(napi_env env, const char* message) {
    Napi::TypeError::New(env, message).ThrowAsJavaScriptException();
}
```

Reply:
@vadim-termius, you are right -- I missed this in the conversion.  https://github.com/atom/node-keytar/pull/270 will resolve this.